pip/utils/__init__.py

```diff
@@ -3,7 +3,9 @@
 import contextlib
 import errno
 import locale
-import logging
+# we have a submodule named 'logging' which would shadow this if we used the
+# regular name:
+import logging as std_logging
```

Comment:
It may not be needed anymore, but it's a good idea in any case. The problem
is that this is utils/__init__.py, so as soon as someone else imports
utils/logging.py, any global variable that we have named 'logging' will get
clobbered by the import system, because this *is* utils.logging. In fact
this was happening before, but no one had noticed because 'logging' was
only accessed during initial startup, never during runtime, and the
clobbering happened but not until after the last time it was used. Probably
the real solution is to move all this code into a proper submodule instead
of keeping it in __init__.py, but one thing at a time...
On Nov 6, 2015 4:58 AM, "Donald Stufft" <notifications@github.com> wrote:

> In pip/utils/__init__.py
> <https://github.com/pypa/pip/pull/3224#discussion_r44133422>:
>
> > @@ -3,7 +3,9 @@
> >  import contextlib
> >  import errno
> >  import locale
> > -import logging
> > +# we have a submodule named 'logging' which would shadow this if we used the
> > +# regular name:
> > +import logging as std_logging
>
> I'm confused why this change is needed. You don't seem to be using the
> name logging anywhere except for the stdlib logging. There is a
> pip.utils.logging module, but that shouldn't cause any problems.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pypa/pip/pull/3224/files#r44133422>.
>